Expand About Biome topic

chenette opened the pull request:
Add detailed Biome overview, architecture diagram, examples, and reference info

Signed-off-by: Anne Chenette <chenette@bitwise.io>

vaporos commented on docs/concepts/about_biome.md:
```diff
@@ -1,17 +1,415 @@
+For Rust developers, the `biome` module in the `libsplinter` library provides
+provides the following submodules, which operate independently:
+
+* [`credentials`](https://docs.rs/splinter/0.3.15/splinter/biome/credentials/index.html):
```
The version in the URL is going to get outdated really quickly.

chenette replied:
Alas, yes.  I will research if it's possible to provide a "release-generic" link to a specific submodule's index.html.

chenette replied:
Changed to `/latest/` instead of `/0.3.15/`.  Note that these direct-to-content links are  used only for major items like modules and top-level submodules that are named in the docs.

chenette replied:
Changed to include `/latest` in the path instead of the version number.

chenette commented:
> We should use SVG for images if possible.

Changed from PNG to SVG images.  Also resized the huge DB schema diagram.

davececchi commented on docs/concepts/about_biome.md:
```diff
@@ -1,17 +1,415 @@
+### The libsplinter biome module
+
+For Rust developers, the `biome` module in the `libsplinter` library provides
+provides the following submodules, which operate independently:
```
```suggestion
the following submodules, which operate independently:
```

chenette replied:
Oops - fixed.

chenette commented on docs/concepts/about_biome.md:
```diff
@@ -1,17 +1,415 @@
-REST API module and can be integrated into an existing splinter REST API using
-libsplinter’s rest-api feature.
+Biome is the Splinter component for user, credential, and private key
+management. It isolates user-based information from other Splinter components,
```
Changed wording as suggested.

chenette commented on docs/concepts/about_biome.md:
```diff
@@ -1,17 +1,415 @@
+Biome is the Splinter component for user, credential, and private key
+management. It isolates user-based information from other Splinter components,
+which are entirely key-based &mdash; for example, keys are used to sign
+transactions on behalf of the users who submitted them. Biome stores user and
```
Fixed the wording.

chenette commented on docs/concepts/about_biome.md:
```diff
@@ -1,17 +1,415 @@
-username and password. Not recommended for use in production.
+Biome provides the following functions:
+
+* User management: Creates and stores users by username and ID. Biome generates
```
Changed to "user account management" and added info on what a Splinter user is.

chenette commented:
Updated doc (with several force-pushes) to retitle/reorganize the examples as @davececchi suggested and to provide an overview of the "workflow story". Also added a "Quick links" TOC at the beginning to make the endpoint examples easier to find.  Also made several corrections and wording improvements. 

Thanks, Dave, for the helpful review!

chenette commented:
The recent force-pushed updates  emove the "Quick links" section, because the published docs now have a right-nav menu for navigating within a topic.

Plus fixes for doc links and a handful of wording issues.

davececchi commented on docs/0.4/concepts/about_biome.md:
```diff
@@ -1,17 +1,422 @@
+
+* Credential management: Registers and authenticates users.  The initial
+  implementation uses usernames and passwords for authentication (not
+  recommended for production use).
```
Let's clarify what exactly we do not recommend using in production.

peterschwarz replied:
Let’s expand on this in a future PR

davececchi replied:
Or maybe just say "The initial implementation uses usernames and passwords for authentication and is not recommended for production usage"?  Like we're not saying credential management in Biome isn't recommended in PROD, we're just saying the first implementation isn't.

chenette replied:
Made this change.

davececchi commented on docs/0.4/concepts/about_biome.md:
```diff
@@ -1,17 +1,422 @@
+
+The design of Biome includes the following assumptions:
+
+* The front-end application is responsible for encrypting the user's private key
```
Is this more of a "client application" than a "front end"?

chenette replied:
Made this change.
